src/utils/v2/distributions.ts

```diff
@@ -0,0 +1,96 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { Split } from 'models/v2/splits'
+
+import { formatSplitPercent, MAX_DISTRIBUTION_LIMIT } from './math'
+
+/**
+ * Gets distribution amount from percent of the distribution limit and then applies
+ * the protocol fee
+ * @param percent {float} - value as a percentage.
+ * @param distributionLimit string (hexString)
+ * @param feePercentage string (hexString)
+ * @returns {number} distribution amount
+ */
+export function getDistributionAmountFromPercentAfterFee({
+  percent,
+  distributionLimit,
+  feePercentage,
+}: {
+  percent: number
+  distributionLimit: string | undefined
+  feePercentage: string
+}) {
+  const amountBeforeFee = getDistributionAmountFromPercentBeforeFee({
+    percent,
+    distributionLimit,
+  })
+
+  if (!amountBeforeFee) return
+
+  return parseFloat(
+    (
+      amountBeforeFee -
+      (amountBeforeFee * parseFloat(feePercentage)) / 100
+    ).toFixed(4),
+  )
+}
+
+/**
+ * Gets distribution amount from percent of the distribution limit and does not apply
+ * any fee
+ * @param percent {float} - value as a percentage.
+ * @param distributionLimit string (hexString)
+ * @returns {number} distribution amount
+ */
+export function getDistributionAmountFromPercentBeforeFee({
```

Comment:
**suggestions**: these functions are really just generic math functions that don't concern themselves with business logic (i.e. distributionLimit)

This naming feels a bit verbose - wdyt about making this a generic `percentOf` function:

```js
const percentOf: (value: number, percent: number) => parseFloat(
    ((percent / 100) * value).toFixed(8)
  )
```

I know we do it in other places, but we should avoid any `undefined`-related logic in these functions. These functions should be as simple as possible.

Reply:
Yeah good call. Changed the function name mentioned and removed undefined's from those functions